src/main/java/net/snowflake/client/core/arrow/tostringhelpers/ArrowArrayStringRepresentationBuilder.java

```diff
@@ -0,0 +1,27 @@
+package net.snowflake.client.core.arrow.tostringhelpers;
+
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+import net.snowflake.client.jdbc.SnowflakeType;
+
+@SnowflakeJdbcInternalApi
+public class ArrowArrayStringRepresentationBuilder extends ArrowStringRepresentationBuilderBase {
+
+  public ArrowArrayStringRepresentationBuilder() {
+    super();
+    this.append('[');
+  }
+
+  public ArrowStringRepresentationBuilderBase appendValue(String value, SnowflakeType valueType) {
```

Comment:
Maybe we should move the type to the constructor? Arrays typically consist of a single type, so this shouldn't cause any issues in the future.

Reply:
makes sense, let me see how it will work out with this append moved to a base class as mentioned in another comment